homeassistant/components/simplisafe/camera.py

```diff
@@ -0,0 +1,240 @@
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up SimpliSafe cameras based on a config entry."""
+    simplisafe = hass.data[DOMAIN][entry.entry_id]
+
+    cameras: list[SimplisafeMotionCamera] = []
+
+    for system in simplisafe.systems.values():
+        if system.version == 2:
+            LOGGER.info("Skipping camera setup for V2 system: %s", system.system_id)
+            continue
+
+        cameras = [
+            SimplisafeMotionCamera(hass, simplisafe, system, camera)
+            for camera in system.cameras.values()
+            if camera.camera_type == CameraTypes.OUTDOOR_CAMERA
+        ]
+
+    async_add_entities(cameras)
+
+
+class SimplisafeMotionCamera(SimpliSafeEntity, Camera):
+    """A camera base class that supports motion capture media."""
+
+    _device: DeviceV3
+    _attr_image_last_updated: datetime | None = None
+    _attr_image_url: str | None = None
+    _attr_clip_url: str | None = None
+    _attr_hls_url: str | None = None
+    _attr_cached_image: bytes | None = None
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        simplisafe: SimpliSafe,
+        system: SystemV3,
+        camera: DeviceV3,
+    ) -> None:
+        """Initialize."""
+        SimpliSafeEntity.__init__(
+            self,
+            simplisafe,
+            system,
+            device=camera,
+            additional_websocket_events=[EVENT_CAMERA_MOTION_DETECTED],
+        )
+        Camera.__init__(self)
+
+        self._hass = hass
+        self._register_services()
+
+    @property
+    def name(self) -> str | None:
+        """Return a good name for this camera."""
+        return self._device.name
+
+    @callback
+    def async_unload(self) -> None:
+        """Release resources."""
+        for service in SERVICES:
+            self._hass.services.async_remove(DOMAIN, service)
+
+    async def async_camera_motion_image(
+        self, width: int | None = None, height: int | None = None
+    ) -> bytes | None:
+        """Return bytes of camera image."""
+        if self._attr_cached_image is not None:
+            return self._attr_cached_image
+
+        if self._attr_image_url is None:
+            return None
+
+        self._attr_cached_image = await self._simplisafe.async_media_request(
+            self._attr_image_url.replace("{&width}", "&width=" + str(width))
+        )
+        return self._attr_cached_image
+
+    def _register_services(self) -> None:
+        """Register the Outdoor Camera hass service calls."""
+
+        def _write_image(to_file: str, image_data: bytes | None) -> None:
+            """Executor helper to write image."""
+            if image_data is None:
+                return
+            os.makedirs(os.path.dirname(to_file), exist_ok=True)
+            with open(to_file, "wb") as img_file:
+                img_file.write(image_data)
+
+        async def save_image_handler(call: ServiceCall) -> None:
```

Comment:
Don't nest this—make this service a top-level one inside this object, then do the service registration in `async_setup_entry`. That'll be consistent with other integrations.